hal/common/critical.c

```diff
@@ -233,10 +233,10 @@ bool core_util_atomic_cas_u32(uint32_t *ptr, uint32_t *expectedCurrentValue, uin
 }
 
 bool core_util_atomic_cas_ptr(void **ptr, void **expectedCurrentValue, void *desiredValue) {
```

Comment:
These new functions will either need to be moved out of the #ifdef to a common area, or implemented for non-m0 targets.